bin/lint-php-i18n.sh

```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+DEFAULT_BRANCH=$(git symbolic-ref refs/remotes/origin/HEAD | sed 's@^refs/remotes/origin/@@')
+CHANGED_PHP_FILES=$(git diff --name-only origin/$DEFAULT_BRANCH | grep '\.php$' | xargs)
```

Comment:
I am seeing some weirdness in my testing as a result of some super-subtle, implicit assumptions in this code:
1. The changes here may include deleted files, which can't be found by `phpcs`. We can work around that by adding `--diff-filter=d` to exclude deleted files.
```suggestion
CHANGED_PHP_FILES=$(git diff --name-only --diff-filter=d origin/$DEFAULT_BRANCH | grep '\.php$' | xargs)
```
2. This code will only work when the current branch has been rebased off of `origin/master`. I am hitting some unexpected files in the command because the current code was branched off from master at some point in the past, but our diff is against the _current_ state of `origin/master`. I think we need to use an interim step to find the right starting point for our diff. I think something like the following should work (based on a quick inspection of https://github.com/sirbrillig/phpcs-changed/blob/trunk/PhpcsChanged/UnixShell.php, which implements a Git-aware `phpcs-changed` command):
```suggestion
INITIAL_COMMIT_FOR_BRANCH=$(git merge-base origin/$DEFAULT_BRANCH HEAD)
CHANGED_PHP_FILES=$(git diff --name-only --diff-filter=d $INITIAL_COMMIT_FOR_BRANCH | grep '\.php$' | xargs)
```
We could also just add `@sirbrillig/phpcs-changed` as a dev dependency to avoid reinventing this wheel!

Reply:
Thanks for the insight! I was unaware of all of these bugs. Will read up on `@sirbrillig/phpcs-changed`

Reply:
Good idea.